contracts/src/TeleporterTokenDestination.sol

```diff
@@ -488,12 +495,24 @@ abstract contract TeleporterTokenDestination is
         // Deposit the funds sent from the user to the bridge,
         // and set to adjusted amount after deposit
         amount = _deposit(amount);
+
+        // The bridged amount must cover the secondary fee, because the secondary fee
+        // is directly subtracted from the bridged amount on the intermediate chain
+        // performing the multi-hop, before forwarding to the final destination chain.
         require(
-            amount > primaryFee + secondaryFee,
-            "TeleporterTokenDestination: insufficient amount to cover fees"
+            amount > secondaryFee, "TeleporterTokenDestination: insufficient amount to cover fees"
         );
 
-        amount -= primaryFee;
+        // Transfer the primary fee to pay for Teleporter fees on the first hop.
+        // The user can specify the destination bridge contract as `feeTokenAddress`,
```

Comment:
```suggestion
        // The user can specify this contract as `feeTokenAddress`,
```

Regarding the TODO, in the case when `feeTokenAddress == address(this)`, it may still be difficult to handle the deposit in a single call to `_deposit` batched in with the `amount` above because in the event that the return value isn't equal to the amount input (i.e. fee/burn on transfer implementations), then it's not clear how much should be used as the bridge amount and how much should be used as the primary fee amount. I do think it's worth avoiding the external contract calls used `safeTransferFrom` in this case though, and just making a call second call to `_deposit`.